### PR TITLE
Update error handling for VersionedLayerClient and VolatileLayerClient

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
* Add error handler for incorrect version and layerId parameters for VersionedLayerClient
* Add error handler for incorrect layerId parameter for VersionedLayerClient
* Update unit-tests

Resolves: OLPSUP-8931
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts [hunk 1]
```diff
@@ -235,16 +240,21 @@ export class VersionedLayerClient {
...
                 version: `${latestVersion}`,
                 layerId: this.layerId,
                 partition: [partitionId],
                 billingTag
             }
-        );
-        const partition = partitions.partitions.find(element => {
+        ).catch(async error => Promise.reject(error));
+
+        if (!partitionsList || !partitionsList.partitions) {
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts]: Please update return model in data-service-api and so on..

Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts]: Updated

File: @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts [hunk 2]
```diff
@@ -109,11 +109,18 @@ export class VersionedLayerClient {
...
-                    quadTreeIndex.subQuads[0].dataHandle,
-                    abortSignal,
-                    dataRequest.getBillingTag()
-                );
+
+                if (quadTreeIndex.status && quadTreeIndex.status === 400) {
+                    return Promise.reject(quadTreeIndex);
+                }
+
+                return quadTreeIndex.subQuads
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts]: what the quadTreeIndex will be if getPartitions will reject error?

Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts]: It will be rejected in the line #123.

**Pull request merged**